### ISSUE TITLE
Newly shared passwords stuck in syncronization

### ISSUE THREAD

**Issue by admorgan** [marius-wieschollek/passwords#323]
### System Information
**Server:**
```
{
    "version": {
        "server": "20.0.4.0",
        "app": "2021.1.0",
        "php": "7.4.11"
    },
    "environment": {
        "os": "Linux",
        "architecture": "i686",
        "bits": 32,
        "database": "pgsql",
        "cron": "cron",
        "proxy": false,
        "sslProxy": false,
        "subdirectory": false
    },
    "legacyApi": {
        "enabled": 3,
        "used": false
    },
    "services": {
        "images": "gdlib",
        "favicons": "bi",
        "previews": "default",
        "security": "hibp",
        "words": "leipzig",
        "previewApi": false,
        "faviconApi": false
    },
    "settings": {
        "channel": "stable",
        "nightlies": false,
        "handbook": false,
        "performance": 1
    },
    "encryption": {
        "sse": {
            "SSEv1r1": false,
            "SSEv1r2": true,
            "SSEv2r1": false,
            "none": false,
            "default": "SSEv1r2"
        },
        "cse": {
            "CSEv1r1": false,
            "none": true,
            "default": "none"
        }
    }
}
```

**Client:**
```
 Browser and Version: Multiple both chrome based and Firefox
 Client OS and Version: OpenSUSE Linux
```

### Steps to reproduce
1. Share a password with another user

### Expected result
Password is shared.

### Actual result
Continuous spinning circle around share with a mouse over stating "Some data is waiting to be synchronized"
Each time I try to share a password an expired session message shows up in the log.
The info log has messages about cron running in global mode

### Nextcloud log
<details>
<summary>Nextcloud log</summary>

```
{"reqId":"Bp6meI3zjHszG9dePykn","level":2,"time":"2021-01-05T21:58:33+00:00","remoteAddr":"192.168.1.1","user":"christine","app":"passwords","method":"GET","url":"/index.php/apps/passwords/api/1.0/service/preview/app.astraltabletop.com/desktop/640/360...","message":"Attempt to access expired or nonexistent session bbd8d6ca-9cfe-4372-9ba6-b84d083db943 by christine","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.107 Safari/537.36","version":"20.0.4.0","id":"5ff4e19e7e174"}
{"reqId":"uYwMRdmmgGcXImZzWwYw","level":1,"time":"2021-01-05T21:59:02+00:00","remoteAddr":"192.168.1.1","user":"christine","app":"passwords","method":"GET","url":"/index.php/apps/passwords/cron/sharing","message":"Passwords runs /index.php/apps/passwords/cron/sharing in global mode","userAgent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.107 Safari/537.36","version":"20.0.4.0","id":"5ff4e19e73186"}
```
</details>

### Browser log
<details>
<summary>Browser log</summary>

```
No errors in the browser log
```
</details>

**Comment by mario-van-zadel** [comment]
I'm having the same issue. The spinning circle doesn't disappear. The cron.php is executed every 5 minutes and produces no error. No errors and warnings in the nextcloud.log (except the one already mentioned by @admorgan):

> Passwords runs /index.php/apps/passwords/cron/sharing in global mode

My system config:

```json
{
    "version": {
        "server": "20.0.4.0",
        "app": "2021.1.0",
        "php": "7.4.10"
    },
    "environment": {
        "os": "Linux",
        "architecture": "x86_64",
        "bits": 64,
        "database": "mysql",
        "cron": "cron",
        "proxy": false,
        "sslProxy": false,
        "subdirectory": false
    },
    "legacyApi": {
        "enabled": 0,
        "used": false
    },
    "services": {
        "images": "imagick",
        "favicons": "default",
        "previews": "default",
        "security": "hibp",
        "words": "random",
        "previewApi": false,
        "faviconApi": false
    },
    "settings": {
        "channel": "stable",
        "nightlies": false,
        "handbook": false,
        "performance": 5
    },
    "sharing": {
        "shares": 10
    },
    "encryption": {
        "sse": {
            "SSEv1r1": false,
            "SSEv1r2": true,
            "SSEv2r1": false,
            "none": false,
            "default": "SSEv1r2"
        },
        "cse": {
            "CSEv1r1": false,
            "none": true,
            "default": "none"
        }
    }
}
```

The request to `/index.php/apps/passwords/cron/sharing` returns the JSON `{"success":false}`.

The request to `/index.php/apps/passwords/api/1.0/password/show` contains the following fields:

```json
{
  ...
  "created":1609910401,
  "updated":1609910401,
  "expires":null,
  "editable":true,
  "shareable":false,
  "updatePending":true
  ...
}
```

**Comment by manu-p** [comment]
Hi, we're facing the same symptom, and in nextcloud.log I find:
`"message":{"Exception":"Exception","Message":"Pageres not found or not accessible","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/passwords/lib/Helper/Preview/PageresCliHelper.php","line":42,"function":"getPageresBinary","class":"OCA\\Passwords\\Helper\\Preview\\PageresCliHelper","type":"->"}`

Installed pageres as explained [here](https://git.mdns.eu/nextcloud/passwords/wikis/Administrators/App-Settings#website-preview-service) and now, the log shows:
`"message":“Attempt to access expired or nonexistent session 5a17a361-1c5a-47f4-ab10-9bd794c0e484 by myUser”`

**Comment by Burschi500** [comment]
Same for me with the following log entry:
`{"reqId":"WrYymFJeE93DCZIP6PLt","level":3,"time":"2021-01-08T14:15:03+01:00","remoteAddr":"","user":"--","app":"core","method":"","url":"--","message":{"Exception":"TypeError","Message":"Typed property OCA\\Passwords\\Db\\AbstractEntity::$userId must be string, null used","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/public/AppFramework/Db/Entity.php","line":151,"function":"setter","class":"OCA\\Passwords\\Db\\AbstractEntity","type":"->"},{"file":"/var/www/nextcloud/apps/passwords/lib/Services/Object/AbstractModelService.php","line":111,"function":"__call","class":"OCP\\AppFramework\\Db\\Entity","type":"->"},{"file":"/var/www/nextcloud/apps/passwords/lib/Services/Object/PasswordService.php","line":87,"function":"createModel","class":"OCA\\Passwords\\Services\\Object\\AbstractModelService","type":"->"},{"file":"/var/www/nextcloud/apps/passwords/lib/Services/Object/AbstractModelService.php","line":80,"function":"createModel","class":"OCA\\Passwords\\Services\\Object\\PasswordService","type":"->"},{"file":"/var/www/nextcloud/apps/passwords/lib/Cron/SynchronizeShares.php","line":194,"function":"create","class":"OCA\\Passwords\\Services\\Object\\AbstractModelService","type":"->"},{"file":"/var/www/nextcloud/apps/passwords/lib/Cron/SynchronizeShares.php","line":112,"function":"createNewShares","class":"OCA\\Passwords\\Cron\\SynchronizeShares","type":"->"},{"file":"/var/www/nextcloud/apps/passwords/lib/Cron/AbstractTimedJob.php","line":62,"function":"runJob","class":"OCA\\Passwords\\Cron\\SynchronizeShares","type":"->"},{"file":"/var/www/nextcloud/lib/private/BackgroundJob/Job.php","line":52,"function":"run","class":"OCA\\Passwords\\Cron\\AbstractTimedJob","type":"->"},{"file":"/var/www/nextcloud/lib/private/BackgroundJob/TimedJob.php","line":59,"function":"execute","class":"OC\\BackgroundJob\\Job","type":"->"},{"file":"/var/www/nextcloud/cron.php","line":127,"function":"execute","class":"OC\\BackgroundJob\\TimedJob","type":"->"}],"File":"/var/www/nextcloud/apps/passwords/lib/Db/AbstractEntity.php","Line":84,"CustomMessage":"Error while running background job (class: OCA\\Passwords\\Cron\\SynchronizeShares, arguments: )"},"userAgent":"--","version":"20.0.4.0","id":"5ff8634c9fc41"}`

**Comment by marius-wieschollek** [comment]
@Burschi500 yes, that is the actual error that causes this issue. I do have a fix for this issue in the current nightly, but because of a different issue related to the distribution of updates i can not release a fixed version before the release of Nextcloud 20.0.5

**Comment by admorgan** [comment]
I can confirm that this fixes my issue.

**Comment by Burschi500** [comment]
How did you fix it, since the nightly is not distributed?

**Comment by admorgan** [comment]
I applied https://github.com/marius-wieschollek/passwords/commit/6031e89a0d90886e64f04d10f2ecb8e78e7a770a to my server

**Comment by Burschi500** [comment]
Thanks, works for me now too.

**Comment by manu-p** [comment]
Hi, still doesn't work for me...

Just upgraded NC 20.0.5 and found Passwords 2020.12.1. 

I'm still experiencing what @admorgan described:
- Continuous spinning circle around share with a mouse over stating "Some data is waiting to be synchronized"
- Each time I try to share a password an expired session message shows up in the log.

Then I applied 6031e89 to my server as mentioned above.

Don't know what I can do to make it work again...

**Comment by marius-wieschollek** [comment]
@manu-p 2020.12.1 is not affected by this bug. If you're having problems with that version, there is probably some other issue.

**Comment by manu-p** [comment]
Thanks @marius-wieschollek. So what is the issue?

As of the original post, I see a continuous spinning circle around share with a mouse over stating "Some data is waiting to be synchronized", and here's what I find in nextcloud.log:

`{"reqId":"At53RgJDAfXprh2F8wHT","level":2,"time":"2021-01-15T12:42:56+00:00","remoteAddr":"A.B.C.D","user":"Manuel","app":"passwords","method":"GET","url":"/apps/passwords/","message":"Attempt to access expired or nonexistent session 202a8065-0478-4360-8005-a57dc70cc01d by Manuel","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:84.0) Gecko/20100101 Firefox/84.0","version":"20.0.5.2"}`

What information else do you need to solve this problem?

I also did try to unactivate/activate the app, and even uninstall/reinstall it.

**Comment by manu-p** [comment]
And just **right now**, it works fine again! :)

**Comment by admorgan** [comment]
I am just postulating, but I believe you most likely needed cron to clean up the previous mess and that is why it is working now.

**Comment by mario-van-zadel** [comment]
I'm now using version 2021.1.0 and Nextcloud 20.0.5.

I still get the following error:

> Exception: Typed property OCA\Passwords\Db\AbstractEntity::$userId must be string, null used

It seems like the commit https://github.com/marius-wieschollek/passwords/commit/6031e89a0d90886e64f04d10f2ecb8e78e7a770a is not included in version 2021.1.0.

**Comment by mario-van-zadel** [comment]
After upgrading to version `2021.1.1-build3852` (nightly build), the sharing of passwords is working again.
I also ran the command `occ config:app:delete passwords cron/sharing/time`, but I'm not sure if it was necessary.

For installing the nightly builds, I followed these instructions: https://git.mdns.eu/nextcloud/passwords/wikis/Administrators/Use-Nightlies

@marius-wieschollek, thank you for fixing it and spending your Friday evening with creating a new build :-)

**Comment by marius-wieschollek** [comment]
This should be fixed with 2021.1.2
